state-chain/pallets/cf-threshold-signature/src/mock.rs

```diff
@@ -193,7 +196,12 @@ impl pallet_cf_threshold_signature::Config<Instance1> for Test {
 pub struct MockVaultActivator;
 impl VaultActivator<MockEthereumChainCrypto> for MockVaultActivator {
 	type ValidatorId = <Test as Chainflip>::ValidatorId;
-	fn activate(_new_key: MockAggKey, _maybe_old_key: Option<MockAggKey>) {}
+	fn start_key_activation(
+		_new_key: MockAggKey,
+		_maybe_old_key: Option<MockAggKey>,
+	) -> Option<ThresholdSignatureRequestId> {
+		None
```

Comment:
This suggests we don't have any unit tests that test the new code path? Since the new code path takes the path when this returns Some(). We should probably add a couple. I ran every code path in my brain to confirm this was the case. 

(I added a panic here and unit tests all pass)
```
						if Signature::<T, I>::get(tss_request_id) == AsyncResult::Void {
							panic!("check");
							T::VaultActivator::activate_key();
						};
```

Reply:
we have an integration test for it, I just tried adding the panic and it correctly fails the test.

I am referring to: integration-tests/src/authorities/authority_rotates_with_the_correct_sequence

Reply:
My thinking is that having at least one unit test to isolate this particular flow means that when code changes and something like an integration test (which is harder to ensure is correct) is changed, nothing is missed, and it's caught in the tests